cognee/modules/chunking/text_chunker_with_overlap.py

```
@@ -0,0 +1,124 @@
+
+logger = get_logger()
+
+
+class TextChunkerWithOverlap(Chunker):
+    def __init__(
+        self,
+        document,
+        get_text: callable,
+        max_chunk_size: int,
+        chunk_overlap_ratio: float = 0.0,
+        get_chunk_data: callable = None,
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)